solutions/task_4.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+"""
+TASK: Implement classes "Temperature", "Person" using Python descriptor protocols to execute unittests without errors.
+"""
+import re
+import unittest
+from datetime import datetime
+
+
+class Temperature(object):
+
+    def __init__(self, value=0):
+        self.fahrenheit = value
+
+    def get_celsius(self):
+        return (self.fahrenheit - 32)/1.8
+
+    def set_celsius(self, value):
+        self.fahrenheit = value*1.8 + 32
+
+    celsius = property(get_celsius, set_celsius)
+
+
+class Descriptor(object):
+
+    def __init__(self, attr_name, attr_type):
+        self.attr_name = attr_name
+        self.attr_type = attr_type
+
+    def __get__(self, instance, owner):
+        return instance.__dict__[self.attr_name]
+
+    def __set__(self, instance, attr_value):
+        if self.attr_type != type(attr_value):
```

Comment:
Can you do this comparison in more pretty way?

Reply:
I suppose yes

Reply:
Lock like "no".. try one more time